spec/models/print_visits_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe PrintVisits, type: :model do
+  context 'with validations' do
+    context 'when the date is out of the the given range' do
+      it 'has an error on the field' do
+        subject.check_date('1-1-2017')
+        expect(subject.errors.messages).to include(base: ["Please choose a date within the last six months, or <a href='%{url}'>contact us</a> if you would like to see older visits."])
+      end
+    end
+
+    context 'when the date is within the the given range' do
+      it 'has no error on the field' do
+        subject.check_date(3.months.ago.strftime("%F"))
+        expect(subject.errors.messages).not_to include(base: ["Please choose a date within the last six months, or <a href='%{url}'>contact us</a> if you would like to see older visits."])
```

Comment:
Duplication on links 8 and 15 ... Error message could be assigned to shared variable